blakserv/ccode.c

```diff
@@ -4660,6 +4665,266 @@ int C_RecordStat(int object_id,local_var_type *local_vars,
+                  r_name->resource_val[0],
+                  r_home->resource_val[0],
+                  r_bind->resource_val[0],
+                  c_guild_name,
+                  stat6.v.data,
+                  stat7.v.data,
+                  stat8.v.data,
+                  stat9.v.data,
+                  stat10.v.data,
+                  stat11.v.data,
+                  stat12.v.data,
+                  stat13.v.data);
+            }
+         }
+         break;
+
+      case STAT_PLAYERSUICIDE:
+         if (num_normal_parms != 3)
+         {
+            bprintf("Wrong Number of Paramenters in C_RecordStat() STAT_PLAYERSUICIDE");
+            break;
+         }
+
+         stat1 = RetrieveValue(object_id, local_vars, normal_parm_array[1].type, normal_parm_array[1].value);
+         stat2 = RetrieveValue(object_id, local_vars, normal_parm_array[2].type, normal_parm_array[2].value);
+
+         if (stat1.v.tag != TAG_SESSION)
+         {
+            bprintf("C_RecordStat STAT_PLAYERSUICIDE can't use non-session %i,%i\n", stat1.v.tag, stat1.v.data);
+            return NIL;
+         }
+
+         if (stat2.v.tag != TAG_RESOURCE)
+         {
+            bprintf("Wrong Type of Parameter in C_RecordStat() STAT_PLAYERSUICIDE");
+            break;
+         }
+
+         session = GetSessionByID(stat1.v.data);
+         r_name = GetResourceByID(stat2.v.data);
+
+         if (!session->account->account_id || !r_name || !r_name->resource_val[0])
+         {
+            bprintf("NULL string in C_RecordStat() for STAT_PLAYERSUICIDE");
+         }
+         else
+         {
+
+            MySQLRecordPlayerSuicide(session->account->account_id, r_name->resource_val[0]);
+         }
+
+         break;
+
+      case STAT_GUILD:
+         if (num_normal_parms != 4)
+         {
+            bprintf("Wrong Number of Paramenters in C_RecordStat() STAT_GUILD");
+            break;
+         }
+
+         stat1 = RetrieveValue(object_id, local_vars, normal_parm_array[1].type, normal_parm_array[1].value);
+         stat2 = RetrieveValue(object_id, local_vars, normal_parm_array[2].type, normal_parm_array[2].value);
+         stat3 = RetrieveValue(object_id, local_vars, normal_parm_array[3].type, normal_parm_array[3].value);
+
+         if (stat1.v.tag != TAG_STRING ||
+            stat2.v.tag != TAG_RESOURCE ||
+            stat3.v.tag != TAG_RESOURCE)
+         {
+            bprintf("Wrong Type of Parameter in C_RecordStat() STAT_GUILD");
+            return NIL;
+         }
+
+         snod = GetStringByID(stat1.v.data);
+         if (snod == NULL)
+         {
+            bprintf("C_RecordStat STAT_GUILD, guild string is null");
+            break;
+         }
+
+
+         c_guild_name = snod->data;
+         r_leader = GetResourceByID(stat2.v.data);
+         r_ghall = GetResourceByID(stat3.v.data);
+
+         if (!r_ghall || !r_ghall->resource_val[0])
+         {
+            snod = GetTempString();
```

Comment:
Still a GetTempString() here that should be converted to a default string like the issue with the room/stroke_obj name.